dev-manuals/test-plans/minigames/regex-game-frontend/plans/u-regex-game-5-validating-invalid-quizzes-works.md

```diff
@@ -0,0 +1,40 @@
+# Regex Game: Validating invalid quizzes works (`u.regex-game-5`)
+
+Version: V1.1, 09.08.2022, Improve format and remove copy paste errors \
+Author: Leon Layer, Leon Hofmeister
+
+## Description
+
+When a new quiz is added, it is validated before being displayed,
+i.e. because the regexes don't compile or because there is no correct answer. \
+An invalid quiz logs an error on the browser console.
+
+## Precondition
+
+-
+
+## Postcondition
+
+An error is logged in the console.
+
+## Typical procedure
+
+1. Open the file `data.ts` and insert a new incorrect quiz whose regex doesn't compile as well as one quiz that has no correct answer
+2. Start the game
+3. There is at least one error message in the browser console
+4. The game works as intended
+
+## Alternative procedures
+
+4.1. The game redirects to the end screen \
+4.2. The game notifies the user that the quizzes are invalid \
+4.3. Unparseable quizzes are skipped
```

Comment:
isn't one of these the typical procedure. it would be strange if the quiz just shows up even though it is not possible to solve

Reply:
Yes, this was quite tricky as well:
When I rewrote @leonleoff's original text, I noticed it as well that this is currently solved suboptimal.
He has already been notified and will likely fix it in a later sprint.
He'll have to adapt this test then, but for now I'd like not to merge tests that must always fail because the functionality isn't present yet.